openvm/src/tmp_register_analysis.rs

```diff
@@ -0,0 +1,322 @@
+        | OPCODE_SRA | OPCODE_SLT | OPCODE_SLTU | OPCODE_MUL | OPCODE_MULH | OPCODE_MULHSU
+        | OPCODE_MULHU | OPCODE_DIV | OPCODE_DIVU | OPCODE_REM | OPCODE_REMU => {
+            let writes = [instruction.a.as_canonical_u32()].into_iter().collect();
+            let rs2_is_reg = instruction.e.is_one();
+            let reads = [instruction.b.as_canonical_u32()]
+                .into_iter()
+                .chain(rs2_is_reg.then_some(instruction.c.as_canonical_u32()))
+                .collect();
+            (reads, writes)
+        }
+        // Load instructions
+        OPCODE_LOADW | OPCODE_LOADBU | OPCODE_LOADHU | OPCODE_LOADB | OPCODE_LOADH => {
+            let writes = [instruction.a.as_canonical_u32()].into_iter().collect();
+            let reads = [instruction.b.as_canonical_u32()].into_iter().collect();
+            (reads, writes)
+        }
+        // Store instructions
+        OPCODE_STOREW | OPCODE_STOREH | OPCODE_STOREB => {
+            let writes = vec![];
+            assert!(instruction.f.is_one());
+            let reads = [
+                instruction.a.as_canonical_u32(),
+                instruction.b.as_canonical_u32(),
+            ]
+            .into_iter()
+            .collect();
+            (reads, writes)
+        }
+        // Branch instructions
+        OPCODE_BEQ | OPCODE_BNE | OPCODE_BLT | OPCODE_BLTU | OPCODE_BGE | OPCODE_BGEU => {
+            let writes = vec![];
+            let reads = [
+                instruction.a.as_canonical_u32(),
+                instruction.b.as_canonical_u32(),
+            ]
+            .into_iter()
+            .collect();
+            (reads, writes)
+        }
+        OPCODE_JAL => {
+            let writes = [instruction.a.as_canonical_u32()].into_iter().collect();
+            let reads = vec![];
+            (reads, writes)
+        }
+        OPCODE_JALR => {
+            let writes = [instruction.a.as_canonical_u32()].into_iter().collect();
+            let reads = [instruction.b.as_canonical_u32()].into_iter().collect();
+            (reads, writes)
+        }
+        OPCODE_LUI | OPCODE_AUIPC => {
+            let writes = [instruction.a.as_canonical_u32()].into_iter().collect();
+            let reads = vec![];
+            (reads, writes)
+        }
+        OPCODE_HINT_STOREW => {
+            let writes = vec![];
+            let reads = [instruction.b.as_canonical_u32()].into_iter().collect();
+            (reads, writes)
+        }
+        OPCODE_HINT_BUFFER => {
+            let writes = vec![];
+            let reads = [
+                instruction.a.as_canonical_u32(),
+                instruction.b.as_canonical_u32(),
+            ]
+            .into_iter()
+            .collect();
+            (reads, writes)
+        }
+        // TERMINATE
+        0 => (vec![], vec![]),
+        // PHANTOM
+        1 => {
+            // TODO: Not sure what should happen here.
```

Comment:
https://github.com/powdr-labs/openvm/blob/main/crates/vm/src/arch/segment.rs#L272

Here it doesn't seem to read/write registers. Would need to check other occurrences